### ISSUE TITLE
sampling an Image should set a label

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#2180]
```python
import holoviews as hv
import numpy as np

hv.notebook_extension()

im = hv.Image(np.random.rand(10, 10))
im.sample(x=0) * im.sample(x=0.1)
```
produces
![download](https://user-images.githubusercontent.com/6897215/33772427-e9f836ba-dc34-11e7-8a59-0fa504c28b3e.png)

However, I would expect to see:
![download 1](https://user-images.githubusercontent.com/6897215/33772448-f9b62436-dc34-11e7-86c6-f7d30f4ffa77.png)

**Comment by jlstevens** [comment]
Good suggestion!

**Comment by philippjfr** [comment]
I agree this makes sense, although I think maybe the correct way of handling this might be to revive the idea of constant dimensions or ``cdims``. If the user supplies a custom label I wouldn't expect that to be overridden so the ``label`` might not be the best mechanism for this.
